WebKit::LegacyCustomProtocolManager::didReceiveResponse() should take a specific type for cacheStoragePolicy

ddkilzer opened the pull request:
#### 07e9312ccba7cc1c280a7638c4e03030db0aba06
<pre>
WebKit::LegacyCustomProtocolManager::didReceiveResponse() should take a specific type for cacheStoragePolicy
<a href="https://bugs.webkit.org/show_bug.cgi?id=250006">https://bugs.webkit.org/show_bug.cgi?id=250006</a>
&lt;rdar://63360220&gt;

Reviewed by Alex Christensen.

Update LegacyCustomProtocolManager::didReceiveResponse() to use
new WebKit::CacheStoragePolicy enum class instead of uint32_t.
Helper methods are used to convert between CacheStoragePolicy
and NSURLCacheStoragePolicy enum values.

* Source/WebKit/DerivedSources-input.xcfilelist:
* Source/WebKit/DerivedSources.make:
* Source/WebKit/PlatformMac.cmake:
- Add CacheStoragePolicy.serialization.in to build files.

* Source/WebKit/NetworkProcess/CustomProtocols/Cocoa/LegacyCustomProtocolManagerCocoa.mm:
(WebKit::LegacyCustomProtocolManager::didReceiveResponse):
* Source/WebKit/NetworkProcess/CustomProtocols/LegacyCustomProtocolManager.h:
(WebKit::LegacyCustomProtocolManager::didReceiveResponse):
* Source/WebKit/NetworkProcess/CustomProtocols/LegacyCustomProtocolManager.messages.in:
- Switch to use WebKit::CacheStoragePolicy.

* Source/WebKit/Shared/Cocoa/CacheStoragePolicy.h: Add.
(WebKit::CacheStoragePolicy):
(WebKit::toNSURLCacheStoragePolicy):
(WebKit::toCacheStoragePolicy):
- Add enum class and conversion functions.

* Source/WebKit/Shared/Cocoa/CacheStoragePolicy.serialization.in: Add.
- Add for IPC.

* Source/WebKit/UIProcess/Cocoa/LegacyCustomProtocolManagerClient.mm:
(-[WKCustomProtocolLoader connection:didReceiveResponse:]):
* Source/WebKit/UIProcess/Network/CustomProtocols/LegacyCustomProtocolManagerProxy.cpp:
(WebKit::LegacyCustomProtocolManagerProxy::didReceiveResponse):
* Source/WebKit/UIProcess/Network/CustomProtocols/LegacyCustomProtocolManagerProxy.h:
(WebKit::LegacyCustomProtocolManagerProxy::didReceiveResponse):
- Switch to use WebKit::CacheStoragePolicy.

* Source/WebKit/WebKit.xcodeproj/project.pbxproj:
- Add CacheStoragePolicy.h and
  CacheStoragePolicy.serialization.in to Xcode project.

Canonical link: <a href="https://commits.webkit.org/258433@main">https://commits.webkit.org/258433@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/0e0b1f35d3afb65563cb5fdf9f9b85e4dd78afeb

| Misc | iOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/6/builds/101958 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/77/builds/11103 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/43/builds/35030 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/8/builds/111291 "Built successfully") | 
| | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/76/builds/12071 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/85/builds/2019 "Built successfully") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/36/builds/94365 "Built successfully") | [✅ 🛠 wincairo](https://ews-build.webkit.org/#/builders/12/builds/109045 "Built successfully") 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/19/builds/107739 "Passed tests") | [  ~~🧪 ios-wk2~~](https://ews-build.webkit.org/#/builders/78/builds/9241 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/3/builds/92506 "Passed tests") | [  ~~🧪 gtk-wk2~~](https://ews-build.webkit.org/#/builders/35/builds/37071 "The change is no longer eligible for processing. Commit was outdated when EWS attempted to process it.") | 
| | [  ~~🧪 api-ios~~](https://ews-build.webkit.org/#/builders/9/builds/91121 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🧪 mac-wk1~~](https://ews-build.webkit.org/#/builders/73/builds/23972 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🧪 api-gtk~~](https://ews-build.webkit.org/#/builders/34/builds/78800 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | 
| | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/81/builds/4686 "Built successfully") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/70/builds/25417 "Passed tests") | | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/82/builds/4776 "Built successfully") | [✅ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/84/builds/1857 "Passed tests") | | 
| | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/80/builds/10852 "Built successfully") | [✅ 🧪 mac-wk2-stress](https://ews-build.webkit.org/#/builders/62/builds/44905 "Passed tests") | | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/79/builds/6525 "Built successfully") | | | 
| [✅ 🛠 🧪 unsafe-merge](https://ews-build.webkit.org/#/builders/75/builds/3041 "Built successfully and passed tests") | | | | 
<!--EWS-Status-Bubble-End-->

achristensen07 commented on Source/WebCore/PAL/pal/spi/cf/CFNetworkSPI.h:
```diff
@@ -515,4 +516,49 @@ WTF_EXTERN_C_END
 
+namespace WebCore {
+
+inline NSURLCacheStoragePolicy toNSURLCacheStoragePolicy(CFURLCacheStoragePolicy policy)
```
I don't think this is a good place for this.  An SPI header should be used for declaration of SPI, which this is not.

ddkilzer replied:
Fixed by creating a `WebKit::CacheStoragePolicy` enum class.

achristensen07 commented on Source/WebCore/PAL/pal/spi/cf/CFNetworkSPI.h:
```diff
@@ -29,6 +29,7 @@
 
 #include <CFNetwork/CFNetwork.h>
 #include <pal/spi/cf/CFNetworkConnectionCacheSPI.h>
+#include <wtf/EnumTraits.h>
```
We're trying to move to *.serialization.in instead of EnumTraits.

ddkilzer replied:
Switched to use `*.serialization.in`.  Hope I did it correctly!

achristensen07 commented on Source/WebKit/UIProcess/Network/CustomProtocols/LegacyCustomProtocolManagerProxy.h:
```diff
@@ -25,6 +25,7 @@
 
 #pragma once
 
+#include "CacheStoragePolicy.h"
```
This can probably be a forward declaration instead.
enum class CacheStoragePolicy : uint8_t;

ddkilzer replied:
Done.

achristensen07 commented on Source/WebKit/NetworkProcess/CustomProtocols/LegacyCustomProtocolManager.h:
```diff
@@ -25,6 +25,7 @@
 
 #pragma once
 
+#include "CacheStoragePolicy.h"
```
ditto

ddkilzer replied:
Done.

Commit:
WebKit::LegacyCustomProtocolManager::didReceiveResponse() should take a specific type for cacheStoragePolicy
https://bugs.webkit.org/show_bug.cgi?id=250006
<rdar://63360220>

Reviewed by Alex Christensen.

Update LegacyCustomProtocolManager::didReceiveResponse() to use
new WebKit::CacheStoragePolicy enum class instead of uint32_t.
Helper methods are used to convert between CacheStoragePolicy
and NSURLCacheStoragePolicy enum values.

* Source/WebKit/DerivedSources-input.xcfilelist:
* Source/WebKit/DerivedSources.make:
* Source/WebKit/PlatformMac.cmake:
- Add CacheStoragePolicy.serialization.in to build files.

* Source/WebKit/NetworkProcess/CustomProtocols/Cocoa/LegacyCustomProtocolManagerCocoa.mm:
(WebKit::LegacyCustomProtocolManager::didReceiveResponse):
* Source/WebKit/NetworkProcess/CustomProtocols/LegacyCustomProtocolManager.h:
(WebKit::LegacyCustomProtocolManager::didReceiveResponse):
* Source/WebKit/NetworkProcess/CustomProtocols/LegacyCustomProtocolManager.messages.in:
- Switch to use WebKit::CacheStoragePolicy.

* Source/WebKit/Shared/Cocoa/CacheStoragePolicy.h: Add.
(WebKit::CacheStoragePolicy):
(WebKit::toNSURLCacheStoragePolicy):
(WebKit::toCacheStoragePolicy):
- Add enum class and conversion functions.

* Source/WebKit/Shared/Cocoa/CacheStoragePolicy.serialization.in: Add.
- Add for IPC.

* Source/WebKit/UIProcess/Cocoa/LegacyCustomProtocolManagerClient.mm:
(-[WKCustomProtocolLoader connection:didReceiveResponse:]):
* Source/WebKit/UIProcess/Network/CustomProtocols/LegacyCustomProtocolManagerProxy.cpp:
(WebKit::LegacyCustomProtocolManagerProxy::didReceiveResponse):
* Source/WebKit/UIProcess/Network/CustomProtocols/LegacyCustomProtocolManagerProxy.h:
(WebKit::LegacyCustomProtocolManagerProxy::didReceiveResponse):
- Switch to use WebKit::CacheStoragePolicy.

* Source/WebKit/WebKit.xcodeproj/project.pbxproj:
- Add CacheStoragePolicy.h and
  CacheStoragePolicy.serialization.in to Xcode project.

Canonical link: https://commits.webkit.org/258433@main
